### ISSUE TITLE
Update Apps in the App Catalog to the most current versions

### ISSUE THREAD

**Issue by jpvajda** [newrelic/nr1-catalog-manager#6]
The following apps are behind the most recent repo versions so we should get these updated using the catalog manager.

|App Name|catalog|current
|-----------------|:------------|:--------
nr1-cloud-optimize|2.7.0|v2.7.1|
nr1-container-explorer|0.8.1|v0.9.2|
nr1-datalyzer|		          0.6.1|	   v0.7.0
nr1-deployment-analyzer|	  0.3.1|    v0.3.9
nr1-event-stream	     |   1.4.0|   v1.4.2
nr1-graphiql-notebook	 |   0.6.0|    v0.6.3
nr1-learn-nrql		     |   1.3.1|     v1.3.3
nr1-metrics-aggregator|	  0.9.11|   v0.9.12
nr1-network-telemetry	 |   0.5.2|	  v0.5.3
nr1-nimbus		         |   1.0.1|	   v1.0.2
nr1-observability-maps|	  0.10.0|	  v0.10.1
nr1-pageview-map 	     |   1.4.0|	  v1.4.5
nr1-status-pages	     |   1.1.2|	  v1.1.2

### AC 

- [x] Read the internal doc on how to use the [catalog manager](https://github.com/newrelic/nr1-catalog-manager)
  > ask the team for a link to the doc
- [x] Ensure each new app version won't introduce breaking changes
- [x] update each app the catalog manager
- [x] merge all necessary PRs so the US version of the applications are bumped in `PROD`
- [x] notify  #help-nr1-platform-service to update the `staging` and `EU` versions of these apps.
- [x] Consider if there are ways to improve or optimize this workflow.

**Comment by zstix** [comment]
Make sure that we use the guids in https://github.com/newrelic/nr1-catalog/blob/master/globals.json

**Comment by aswanson-nr** [comment]
nr1-container-explorer is blocked on https://github.com/newrelic/nr1-container-explorer/issues/62

**Comment by aswanson-nr** [comment]
The validation in the catalog manager stop nr1-deployment-analyzer from upgrading because its preview image does not meet the aspect ratio requirements. This may be an issue with the validation itself, I will continue to look into it.

**Comment by aswanson-nr** [comment]
|App Name|catalog|current|status|
|-----------------|:------------|:--------|:--------
nr1-cloud-optimize|2.7.0|v2.7.1| merged
nr1-container-explorer|0.8.1|v0.9.4| merged, appears to have been updated but not published to the catalog https://github.com/newrelic/nr1-catalog/runs/2273124779?check_suite_focus=true
nr1-datalyzer|		          0.6.1|	   v0.7.1| merged, but did not update
nr1-deployment-analyzer|	  0.3.1|    v0.3.11| merged
nr1-event-stream	     |   1.4.0|   v1.4.2| merged, but did not update
nr1-graphiql-notebook	 |   0.6.0|    v0.6.3| blocked by https://github.com/newrelic/nr1-graphiql-notebook/issues/58
nr1-learn-nrql		     |   1.3.1|     v1.3.3| merged
nr1-metrics-aggregator|	  0.9.11|   v0.9.13| is already at 0.9.13
nr1-network-telemetry	 |   0.5.2|	  v0.5.3| merged
nr1-nimbus		         |   1.0.1|	   v1.0.2| merged
nr1-observability-maps|	  0.10.0|	  v0.10.1| merged
nr1-pageview-map 	     |   1.4.0|	  v1.4.5| merged
nr1-status-pages	     |   1.1.2|	  v1.1.2| is already at 1.1.2

I will continue to update this table as I progress.

**Comment by aswanson-nr** [comment]
Other apps updated:
* nr1-github  | V0.6.7
* nr1-tag-improver | V1.1.1

**Comment by aswanson-nr** [comment]
Observations:
* The current process requires a lot of toil. Pulling down the nerdpack, manually testing it, using the `nr1-catalog-manager`, merging into the `nr1-catalog`, and then notifying platform services. Breakages at any point require creating a new release of the nerdpack and starting at the beginning.
* Quick improvements could be made to test and validate the nerdpack before creating a release, removing the need to do this before adding it to the `nr1-catalog`. This should also verify that the catalog requirements are met, such as screenshot aspect ratio and size.
* After the improvements above, the `nr1-catalog-manager` could be phased out in favor of the nerdpack itself controlling releases to the `nr1-catalog` repository.
* Another option is to remove the `nr1-catalog` and the `nr1-catalog-manager`, allowing each nerdpack to handle its own release process.
* If the `nr1-catalog` must stick around to manage the current prod versions of the nerdpacks, then moving it away from git submodules and over to a semantic version based system (think `package.json` dependencies) will reduce a lot of headaches. Notably avoiding issues around merging a new PR before an older one.

@jpvajda Would you like me to move these to a separate issue? I'm not quite sure where they should go

**Comment by zstix** [comment]
It looks like the only remaining acceptance criteria on this ticket is:
> Consider if there are ways to improve or optimize this workflow.

You've outlined some great options here, which I think _could_ satisfy that last requirement. We might want to make a separate ticket for a research spike on how we can improve this process & document that for the community. I suspect that this will have a hard time getting into the roadmap, since we have a few other things we're focusing on.

**Comment by jpvajda** [comment]
adding @polfliet for visibility in this assessment. 

Thanks @aswanson-nr!  - There was a decent amount of development time that went into the catalog manager, so I'd like to first sync with the creators of, namely @jbeveland27 for input on our findings and some of the recommendations we are coming up with to move forward with our goal to support 3rd Party Apps with little to  no effort from our team.

Perhaps we can take this input and schedule some time with Justin to talk through it, I bet he can shed some light on what maybe didn't get done, that he intended to do, prior to be moved to another project. I'd like to that before we decide to move away from that particular solution that is already in place.

I'm not saying this is the right / wrong I just think I need a bit more context before knowing how to respond here. 😅

**Comment by aswanson-nr** [comment]
Great idea! Yeah, these findings are just from my usage of the `nr1-catalog-manager` over the last week. I don't know anything about the original requirements for this thing. I spoke with @jbeveland27 yesterday and he did mention that there was originally more development planned.

**Comment by zstix** [comment]
We're going to close this issue out, since the original acceptance criteria is (mostly) complete. We're going to file new tickets for the remaining apps that need updates and will continue the conversation about how we would like to move forward with this ecosystem.

**Comment by jpvajda** [comment]
https://github.com/newrelic/nr1-catalog/issues/232
